classes/stock/StockAvailable.php

```diff
@@ -646,22 +646,30 @@ public static function removeProductFromStockAvailable($id_product, $id_product_
             return false;
         }
 
-        if (Shop::getContext() == Shop::CONTEXT_SHOP) {
-            if (Shop::getContextShopGroup()->share_stock == 1) {
-                $pa_sql = '';
-                if ($id_product_attribute !== null) {
-                    $pa_sql = '_attribute';
-                    $id_product_attribute_sql = $id_product_attribute;
-                } else {
-                    $id_product_attribute_sql = $id_product;
-                }
+        if (null !== $shop) {
+            if (!($shop instanceof Shop)) {
+                $shop = new Shop($shop);
+            }
+            $groupSharedStock = $shop->getGroup()->share_stock == 1;
```

Comment:
maybe this is better way? (at least we don't need that loose comparison ('==') which php stan ussualy doesn't like. And I think it works as we expect it?

```suggestion
            $groupSharedStock = (bool) $shop->getGroup()->share_stock;
```